File: server/makc2022--nodejs--nestjs/src/layer3/nosql-mongo/sample/mappers/typegoose/entities/dummy-main/mapper-dummy-main-entity.extension.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { DummyMainEntityLoader } from 'src/layer3/nosql-mongo/sample/entities/dummy-main/dummy-main-entity.loader';
import { DummyMainEntityObject } from 'src/layer3/nosql-mongo/sample/entities/dummy-main/dummy-main-entity.object';
import { MapperDummyMainEntityObject } from './mapper-dummy-main-entity.object';

/** Расширение сущности "DummyMain" сопоставителя. */
export class MapperDummyMainEntityExtension {
  private constructor(
    private self: {
      entityObject?: DummyMainEntityObject;
      mapperObject?: MapperDummyMainEntityObject;
    }
  ) {}

  /** Расширить объект сущности.
   * @param entityObject Объект сущности.
   */
  static extEntityObject(entityObject: DummyMainEntityObject) {
    return new MapperDummyMainEntityExtension({ entityObject });
  }

  /** Расширить объект сопоставителя.
   * @param mapperObject Объект сопоставителя.
   */
  static extMapperObject(mapperObject: MapperDummyMainEntityObject) {
    return new MapperDummyMainEntityExtension({ mapperObject });
  }

  /** Преобразовать из объекта сущности в объект сопоставителя.
   * @returns Объект сопоставителя.
   */
  fromEntityToMapperObject(): MapperDummyMainEntityObject {
    const result = new MapperDummyMainEntityObject();

    new DummyMainEntityLoader(result).load(this.self.entityObject);

    return result;
  }

  /** Преобразовать из объекта сопоставителя в объект сущности.
   * @returns Объект сущности.
   */
  fromMapperToEntityObject(): DummyMainEntityObject {
    const loader = new DummyMainEntityLoader();

    loader.load(this.self.mapperObject);

    return loader.entityObject;
  }
}

File: server/makc2022--dotnet/Makc2022.Layer3.Sql.Sample.Mappers.EF/Entities/DummyMain/MapperDummyMainEntityExtension.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2022.Layer3.Sql.Sample.Entities.DummyMain;

namespace Makc2022.Layer3.Sql.Sample.Mappers.EF.Entities.DummyMain
{
    /// <summary>
    /// Расширение сущности "DummyMain" сопоставителя.
    /// </summary>
    public static class MapperDummyMainEntityExtension
    {
        #region Public methods

        /// <summary>
        /// Преобразовать из объекта сущности в объект сопоставителя.
        /// </summary>
        /// <param name="entityObject">Объект сущности.</param>
        /// <returns>Объект сопоставителя.</returns>
        public static MapperDummyMainEntityObject FromEntityToMapperObject(
            this DummyMainEntityObject entityObject
            )
        {
            MapperDummyMainEntityObject result = new();

            new DummyMainEntityLoader(result).Load(entityObject);

            return result;
        }

        /// <summary>
        /// Преобразовать из объекта сопоставителя в объект сущности.
        /// </summary>
        /// <param name="mapperObject">Объект сопоставителя.</param>
        /// <returns>Объект сущности.</returns>
        public static DummyMainEntityObject FromMapperToEntityObject(
            this MapperDummyMainEntityObject mapperObject
            )
        {
            DummyMainEntityLoader loader = new();

            loader.Load(mapperObject);

            return loader.EntityObject;
        }

        #endregion Public methods
    }
}
